### PR TITLE
to fix #15301

### LINKED ISSUE #15301: please fixed the this code ArrayHelper::filter line 941
A small bug. when i write this : 
    `ArrayHelper::filter(['a'=>0],['a']);`
above code will return empty array to me.
i have check on the ArrayHelper::filter line 941
i saw this code:
   `if (empty($array[$globalKey])) {
                continue;
    }`
i think it modify to follow is better:
    `if (!key_exists($globalKey, $array)) {
                continue;
     }`

### PR DESCRIPTION
**Pull request by hhniao**
| Q             | A
| ------------- | ---
| Is bugfix?    | yes
| New feature?  | no
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | fixed ArrayHelper::filter line 941  #15301 fixed by the PR, if any

### CODE REVIEW

File: framework/helpers/BaseArrayHelper.php [hunk 1]
```diff
@@ -938,7 +938,11 @@ public static function filter($array, $filters)
                 continue;
             }
 
-            if (empty($array[$globalKey])) {
+//            if (empty($array[$globalKey])) {
+//                continue;
+//            }
```
Comment (MysteryDragon) [inline framework/helpers/BaseArrayHelper.php]: I guess it's better to remove code in comments.

**Review by MysteryDragon — COMMENTED** [review]
I'd say that is not so desirable to change code formatting in tests (look at Files changed tab, tests code). Its 'cause of changing PSR-compatible code into not PSR-compatible. Yii 2 follows PSR-2...

File: tests/framework/helpers/ArrayHelperTest.php [hunk 2]
```diff
@@ -1269,5 +1228,19 @@ public function testFilter()
...
+            'a' => 0,
+            'b' => null,
+            'c' => '',
+            'd' => true,
+            'e' => false,
+        ];
+        //before fixed
+        //ArrayHelper::filter($tmp, ['a','b','c','d','e']);
+        //['d'=>true]
+        //after fixed
```
Comment (MysteryDragon) [inline tests/framework/helpers/ArrayHelperTest.php]: I think this comments are also not so necessary

**Comment by MysteryDragon** [comment]
Also it will be required to add information into `framework/CHANGELOG.md` about this fix.

File: tests/framework/helpers/ArrayHelperTest.php [hunk 3]
```diff
@@ -12,50 +13,51 @@
 use yii\helpers\ArrayHelper;
 use yiiunit\TestCase;
 
-class Post1
-{
+class Post1 {
```
Comment (samdark) [inline tests/framework/helpers/ArrayHelperTest.php]: Code should not be reformatted.

**Comment by MysteryDragon** [comment]
I guess this one must be closed in flavor of #15312

**Pull request closed without merging**